### DISCUSSION TITLE
vscode intellisense is super slow but other frameworks is good

### DISCUSSION THREAD

**Discussion by khanakia** [remix-run/remix#9397] [Q&A]
I am using remix.js for one of my projects and vscode IntelliSense is super slow.

I have other projects in Next.js, and Astro they are super fast. 

https://github.com/remix-run/remix/assets/2174170/aebd1147-cd49-4541-b8d4-24400e0f1041

**Comment by kiliman** [comment]
What does your `tsconfig.json` look like? Does it contain `compilerOptions.skipLibCheck: true`?

You may also want to set `compilerOptions.allowJs: false`.

**Reply by khanakia** [reply]
It's default form remix and the project is not that big it's very small project 7-10 pages app

PS: just tried `compilerOptions.allowJs: false` speed is same

```json
{
  "include": [
    "**/*.ts",
    "**/*.tsx",
    "**/.server/**/*.ts",
    "**/.server/**/*.tsx",
    "**/.client/**/*.ts",
    "**/.client/**/*.tsx"
  ],
  "compilerOptions": {
    "lib": ["DOM", "DOM.Iterable", "ES2022"],
    "types": ["@remix-run/node", "vite/client", "google.accounts"],
    "isolatedModules": true,
    "esModuleInterop": true,
    "jsx": "react-jsx",
    "module": "ESNext",
    "moduleResolution": "Bundler",
    "resolveJsonModule": true,
    "target": "ES2022",
    "strict": true,
    "allowJs": true,
    "skipLibCheck": true,
    "forceConsistentCasingInFileNames": true,
    "baseUrl": ".",
    "paths": {
      "~/*": ["./app/*"]
    },

    // Vite takes care of building everything, not tsc.
    "noEmit": true
  }
}
```

**Comment by khanakia** [comment]
@kiliman i have created a very basic repo by removing everything even with just one file still slow
 https://github.com/khanakia/testremix
```
yarn
```

You can test it also I tested with github codespaces just to make sure

**Comment by khanakia** [comment]
It seems the remix is watching the `node_modules` directory I just enabled the `tserver` verbose logging in vscode

```
Info 10009[21:15:14.690] Elapsed:: 0.022625088691711426ms DirectoryWatcher:: Triggered with /Volumes/D/testremix/node_modules
```
